java/sage/Stringer.java

```diff
@@ -61,7 +63,37 @@ public int compareTo(Stringer o)
     return toString().compareTo(o.toString());
   }
 
+  // NOTE: Any string passed into this method must already be all lowercase.
+  public boolean equalsIgnoreCase(String compare)
+  {
+    String thisName = name;
+    int thisIgnoreHash = ignoreCaseHash;
+    // The hash could be 0, but it's not going to be any more often than any other hash and in case
+    // we miss a case where the title is set, this covers that problem.
+    if (thisIgnoreHash == 0)
+      return thisName.equalsIgnoreCase(compare);
+    return compare == thisName ||
+      (compare.hashCode() == thisIgnoreHash && compare.length() == thisName.length() &&
```

Comment:
It is marginally faster because we don't match more often than we do. That is especially true when dealing with things like movies and paid programming. Also you might have noticed that this is be called predominantly by String constants, so the hash should only be getting computed once. This method has no purpose without this check. Otherwise we would just use equalsIgnoreCase() directly.